### PR TITLE
Add explicit directional step transitions to WorkflowPaywallView

### PR DESCRIPTION
**Pull request by facumenzella**
## Summary

Builds on top of #6692 (`WorkflowPaywallView`) to add robust directional slide transitions when navigating between workflow steps, and cleans up a few rough edges from the initial implementation.

**What's in this PR:**
- Directional slide transitions between steps driven by explicit transition state rather than SwiftUI `.transition()`
- Workflow-aware back/dismiss behavior during step navigation, while keeping existing built-in close button support
- Guards against double-taps during transitions via `allowsHitTesting(false)`
- Respects `accessibilityReduceMotion` (instant swap, no animation)
- Keeps rendered pages keyed by stable snapshot IDs so SwiftUI preserves the outgoing subtree during transitions and avoids flashing
- Unit tests for the transition state machine

### Why not SwiftUI `.transition()`

SwiftUI captures a view's removal transition at insertion time. When navigating back after going forward, the outgoing view still has the forward removal transition baked in — it slides left instead of right. This breaks on every first change of direction.

The fix is `WorkflowPageTransitionState<Page>`, a small state machine that owns both the current and outgoing page simultaneously and drives the animation via explicit `.offset()` rather than `.transition()`. Direction and offsets are computed at render time from current state, so they're always correct regardless of navigation history.

### Why stable page identity matters

The outgoing page changes role from `current` to `outgoing` during a transition. Keeping each rendered page keyed by a stable snapshot ID lets SwiftUI preserve that subtree instead of recreating it in a different branch of the view tree, which avoids visible flashing at transition start.

### Why `DispatchQueue.main.async` to kick off the animation

When `beginTransition` is called, `progress` is set to `0` — both pages are at their starting offsets (outgoing centered, incoming off-screen). The animation then needs to drive `progress` to `1`.

If `advanceAnimation()` were called immediately in the same synchronous block, SwiftUI would batch both state changes (`progress = 0` then `progress = 1`) into a single render pass and skip straight to the final positions — no animation would play.

Deferring via `DispatchQueue.main.async` lets SwiftUI commit one frame with both pages at their initial offsets first. When the deferred block fires, `withAnimation` has a rendered starting state to animate from, so the slide plays correctly.

https://github.com/user-attachments/assets/1bc0ff5d-a892-4038-96f5-9c62c00c875d

---

> [!NOTE]
> **Medium Risk**
> Touches paywall workflow navigation/rendering and introduces async animation timing, which could affect step changes or dismissal behavior if edge cases slip through; changes are localized and covered by new unit tests.
> 
> **Overview**
> Adds an explicit `WorkflowPageTransitionState` and refactors `WorkflowPaywallView` to render *current* and *outgoing* step pages simultaneously, driving **directional slide transitions** via computed `.offset`/`.zIndex` instead of SwiftUI `.transition()`.
> 
> Navigation/dismiss handling is updated to start forward/back transitions (including guarding against interactions mid-transition with `allowsHitTesting(false)`), respect `accessibilityReduceMotion` by swapping instantly, and keep stable page identity to avoid flashing; new tests validate the transition state machine’s offsets, z-order, and completion behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5c14368246bab892c7b40f5ab58bb65ec72d3488. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: RevenueCatUI/Templates/V2/WorkflowPaywallView.swift [hunk 1]
```diff
@@ -34,6 +143,8 @@ struct WorkflowPaywallView: View {
 
     @StateObject private var navigator: WorkflowNavigator
     @State private var hasLoggedInvalidState = false
+    @State private var transitionState: WorkflowPageTransitionState<RenderedPage>
```
[the remark this answers is not in the document]
Comment (facumenzella) [reply RevenueCatUI/Templates/V2/WorkflowPaywallView.swift:146]: Keeping it as-is for now —  needs to stay alive for the view's lifetime and the extra re-renders are harmless correctness-wise. We'll track it as a follow-up cleanup.

**Review by vegaro — APPROVED** [review]
looks solid as a base PR. The video in the description is not showing the hero image in the first page, is that unrelated?

**Comment by facumenzella** [comment]
@vegaro it's not related, I was having connection issues when I recorded it 🤓
